File: SDKs/shutterstock.com/1.1.5/typescript/src/sdk/models/operations/geteditoriallivefeeditems.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetEditorialLivefeedItemsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class GetEditorialLivefeedItemsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=country" })
  country: string;
}


export class GetEditorialLivefeedItemsSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=http;subtype=basic" })
  basic?: shared.SchemeBasic;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  customerAccessCode?: shared.SchemeCustomerAccessCode;
}


export class GetEditorialLivefeedItemsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetEditorialLivefeedItemsPathParams;

  @SpeakeasyMetadata()
  queryParams: GetEditorialLivefeedItemsQueryParams;

  @SpeakeasyMetadata()
  security: GetEditorialLivefeedItemsSecurity;
}


export class GetEditorialLivefeedItemsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  editorialContentDataList?: any;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/shutterstock.com/1.1.5/python/src/sdk/models/operations/geteditoriallivefeeditems.py
import dataclasses
from typing import Any,Optional
from ..shared import security as shared_security
from ..shared import security as shared_security


@dataclasses.dataclass
class GetEditorialLivefeedItemsPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetEditorialLivefeedItemsQueryParams:
    country: str = dataclasses.field(metadata={'query_param': { 'field_name': 'country', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetEditorialLivefeedItemsSecurity:
    basic: Optional[shared_security.SchemeBasic] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    customer_access_code: Optional[shared_security.SchemeCustomerAccessCode] = dataclasses.field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class GetEditorialLivefeedItemsRequest:
    path_params: GetEditorialLivefeedItemsPathParams = dataclasses.field()
    query_params: GetEditorialLivefeedItemsQueryParams = dataclasses.field()
    security: GetEditorialLivefeedItemsSecurity = dataclasses.field()
    

@dataclasses.dataclass
class GetEditorialLivefeedItemsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    editorial_content_data_list: Optional[Any] = dataclasses.field(default=None)
    
